packages/firebase_database/firebase_database_platform_interface/lib/src/platform_interface/event.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of firebase_database_platform_interface;
+
+/// Enum to define various types of database events
+enum EventType {
+  /// Event for [onChildAdded] listener
+  childAdded,
+
+  /// Event for [onChildRemoved] listener
+  childRemoved,
+
+  /// Event for [onChildChanged] listener
+  childChanged,
+
+  /// Event for [onChildMoved] listener
+  childMoved,
+
+  /// Event for [onValue] listener
+  value,
+}
+
+/// `Event` encapsulates a DataSnapshot and possibly also the key of its
+/// previous sibling, which can be used to order the snapshots.
+class EventPlatform {
```

Comment:
do we really need to create separate model classes for Platform. We can move the Event class for e.g, to the platform_interface (of course with the public constructor) and then we can export this class to use it in the firebase_database for e.g.